manim/animation/movement.py

```diff
@@ -44,6 +44,17 @@ class Homotopy(Animation):
         Keyword arguments propagated to :meth:`.Mobject.apply_function`.
     kwargs
         Further keyword arguments passed to the parent class.
+
+    Examples
+    --------
+    .. manim:: HomotopyExample
+
+        class HomotopyExample(Scene):
+            def construct(self):
+                dot = Dot()
+                homotopy = lambda x,y,z,t:[x + t, y + t**2, z]
+
+                self.play(Homotopy(homotopy, dot))
```

Comment:
I like the suggestion of @sparshg, but dislike that the square is already deformed in the initial step. I have an alternate suggestion of a more classical example of a homotopy:

```suggestion
    .. manim:: HomotopyExample

        import math

        class HomotopyExample(Scene):
            def construct(self):
                square = Square()
                square.insert_n_curves(20)  # improve quality of homotopy target
                homotopy_map = lambda x, y, z, t: (
                    (1-t)*x + t*2*np.cos(math.atan2(y, x)),
                    (1-t)*y + t*2*np.sin(math.atan2(y, x)),
                    z,
                )
                self.play(Homotopy(homotopy_map, square))
```

Reply:
Output:

https://user-images.githubusercontent.com/11851593/201829134-db90993c-5945-41e9-a3f6-951ab7cf530a.mp4

What do you think?